src/utils/filterPeople.ts

```diff
@@ -0,0 +1,47 @@
+import {Person} from "../types";
+
+function isPersonContainsQuery(person: Person, query: string) {
+  const validQuery = query.toLowerCase();
+  const {name, motherName, fatherName} = person;
+
+  return name.toLowerCase().includes(validQuery)
+    || motherName?.toLowerCase().includes(validQuery)
+    || fatherName?.toLowerCase().includes(validQuery);
+}
+
+function isPersonFitsCentury(person: Person, centuries: string[]) {
+  const years = person.died - person.born;
+  const birthYear = person.died - years;
+  const personCentury = String(Math.floor(birthYear / 100) + 1);
+
+  return centuries.includes(personCentury);
+}
+
+export function filterPeople(
+  people: Person[], filterParams: [string, string, string[]],
+) {
+  let filteringPeople = [...people];
```

Comment:
Method `filter` returns a new array every time, so you don’t need to copy it with the spread operator.